### ISSUE TITLE
([Bloodmaul Geomancer][Drop] Formula: Enchant Bracer - Spellpower)

### ISSUE THREAD

**Issue by divinenips** [Atlantiss/NetherwingBugtracker#7832]
Description: Bloodmaul Geomancer in Blade's Edge Mountains is not dropping the enchanting pattern "Formula: Enchant Bracer - Spellpower." This has been confirmed both with my own testing (over 30 hours of farming and over 2000 mobs killed), as well as confirmation from GM Crysis that no one has been eligible for the drop for several days.

Current Behavior: Not dropping at all. Unclear if it's due to it failing to drop due to it not recognizing player profession or due to it simply being removed from the drop table.

Expected Behavior: Given the player has Enchanting as a profession, the pattern should drop at a 2% rate according to Wowwiki: https://www.wowhead.com/item=22534/formula-enchant-bracer-spellpower. In fact, it should be somewhat higher as this drop rate of 2% is factoring in all the people killing these NPCs without having the enchanting skill. This in fact should apply to ALL the trade skill BoP items; that is to say that the drop rates, if they are pulled from Wowwiki, would not be accurate to how frequently they drop if one has the proper profession.

Server Revision: 3578

**Comment by divinenips** [comment]
This also appears to be the case with the Ethereal Nullifiers and "Formula: Enchant Bracer - Defense." Similar thing with the intended drop rate and the actuality of the drop rate. Both the Nullifiers and the Geomancers do share a trait which may be the culprit: their spawn points are variable (meaning their spawn points could spawn other mob types). That may not be the issue, but it does seem to be the common thread between the two mob types.

**Comment by ScottieDOESknow** [comment]
The formulas drop, it's just a super low droprate from NPCs which are also required for kill quests.
The way loot works: spawns with NPC so if non-enchanter kills it, the loot is there but they can't see it and that % drop is basically lost until it spawns on another. So that why sometimes it can can drop after 2 kills or 2 weeks of farming. Pray to RNGeesus.

**Comment by divinenips** [comment]
> The formulas drop, it's just a super low droprate from NPCs which are also required for kill quests.
> The way loot works: spawns with NPC so if non-enchanter kills it, the loot is there but they can't see it and that % drop is basically lost until it spawns on another. So that why sometimes it can can drop after 2 kills or 2 weeks of farming. Pray to RNGeesus.

Two problems with this. 

Firstly, It isn't a super low drop rate. Wowwiki says 2% (which is in and of itself not super low, it should probabilistically drop after around 100 kills). That being said, if the drop tables are populated using Wowwiki's original numbers, they are incorrect. It shouldn't be a 2% drop rate, because that drop rate doesn't factor in all the kills from people who weren't enchanters. This is true for ALL of the BoP pattern drops. You could conservatively say that the drop rate for each of these items should be about 5 times higher to account for all of the folks killing these mobs without the proper professions.

Secondly, the rate is not currently at 2%, there's no way it could be at that drop rate if I have accrued over 2000 kills. Probabilistically, it should have dropped for me several times over at this point. This area has very high activity, mostly from folks farming for this drop. Every single Skirmisher (the other mob type that can spawn on this particular mobs spawn point) and every single Geomancer were on the floor after they spawned. That means this area was being fully farmed for hours upon hours (in my case, I was there for 31 hours, I can't speak from before I got there), without a SINGLE drop according to GM Crysis. For days. There's no way that this is reflective of a 2% drop rate.

TL;DR: this is an issue that actually needs to be fixed.

**Comment by ScottieDOESknow** [comment]
The only "problem" I see is that you won't accept this is a notoriously low droprate which is ultimately based on RNG. ;)
My personal experience from retail TBC to private server TBC: this enchant droprate has actually been consistent even on the more poorly scripted servers who don't use sniffs, etc. It's even consistent with Netherwing (the previous Atlantiss TBC server)

One search on Atlantiss discord would have gave you this info from 2 weeks ago...
![image](https://user-images.githubusercontent.com/43949261/103143697-c4eb2080-46e9-11eb-91bd-3b0efabda098.png)

Here is info from our guild discord, that I could find on this formula

(November)
![image](https://user-images.githubusercontent.com/43949261/103143721-3dea7800-46ea-11eb-994e-7b60e71c40f2.png)

(August)
![image](https://user-images.githubusercontent.com/43949261/103143735-8ace4e80-46ea-11eb-87e0-b8e27e63468e.png)

Besides the above anecdotes, I also remember a few guild members farming it for multiple weeks/hours every day. 
Same holds true for the other low RNG enchants as well. In the end, this formula DOES drop but it's LOW droprate. 
I mean... 2000 kills seems super high to never see one so maybe exaggeration is the issue here?
Either that or you have the worst luck on the server!

TL;DR: Accept the fact this is a low droprate formula and keep grindin'

**Comment by Godmatik** [comment]
RNG is RNG

> Secondly, the rate is not currently at 2%, there's no way it could be at that drop rate if I have accrued over 2000 kills.

You're falling for the "Appeal to probability" fallacy.

**Comment by divinenips** [comment]
> The only "problem" I see is that you won't accept this is a notoriously low droprate which is ultimately based on RNG. ;)
> My personal experience from retail TBC to private server TBC: this enchant droprate has actually been consistent even on the more poorly scripted servers who don't use sniffs, etc. It's even consistent with Netherwing (the previous Atlantiss TBC server)
> 
> One search on Atlantiss discord would have gave you this info from 2 weeks ago...
> ![image](https://user-images.githubusercontent.com/43949261/103143697-c4eb2080-46e9-11eb-91bd-3b0efabda098.png)
> 
> Here is info from our guild discord, that I could find on this formula
> 
> (November)
> ![image](https://user-images.githubusercontent.com/43949261/103143721-3dea7800-46ea-11eb-994e-7b60e71c40f2.png)
> 
> (August)
> ![image](https://user-images.githubusercontent.com/43949261/103143735-8ace4e80-46ea-11eb-87e0-b8e27e63468e.png)
> 
> Besides the above anecdotes, I also remember a few guild members farming it for multiple weeks/hours every day.
> Same holds true for the other low RNG enchants as well. In the end, this formula DOES drop but it's LOW droprate.
> I mean... 2000 kills seems super high to never see one so maybe exaggeration is the issue here?
> Either that or you have the worst luck on the server!
> 
> TL;DR: Accept the fact this is a low droprate formula and keep grindin'

I'll tell you what bro, if you can get this thing to drop for you and provide screenies, I'll pay you $25 on Venmo. Not even joking. Go for it, prove me wrong.

**Comment by divinenips** [comment]
> RNG is RNG
> 
> > Secondly, the rate is not currently at 2%, there's no way it could be at that drop rate if I have accrued over 2000 kills.
> 
> You're falling for the "Appeal to probability" fallacy.

No I'm not. The odds of my not getting the drop if it's on a 2% drop rate are now so low that you are the one committing the appeal to probability fallacy. We are now at over 2800 kills of this mob with no drop. That's not 2%, it's not anywhere near 2%.

**Comment by ScottieDOESknow** [comment]
> I'll tell you what bro, if you can get this thing to drop for you and provide screenies, I'll pay you $25 on Venmo. Not even joking. Go for it, prove me wrong.

I don't need it to drop right now to prove you wrong. There are multiple enchanters in my guild that have had it drop starting from the 2nd week of server up until weeks ago. I even provided evidence from discord with date stamps. Funny thing is, they aren't the only enchanters on the server who have it. There are, most likely, hundreds of enchanters on the server (Horde and Alliance) who have this formula from farming it. Some got it after few kills, others slaved for weeks.  

Seems rather consistent with...

- many Thottbott/WoWHead comments from 2007/2008 
- my recollection personally farming it multiple times 2007/2008 on retail TBC 
- my recollection personally farming it on THREE other TBC private servers in the last 5 years
- my recollection personally farming it in 2018/2019 on TBC Netherwing Atlantiss

So now...You say you've killed 2800 ogres and that seems exaggerated to me, so why don't you provide screenies and data? 
Go for it, prove me wrong.

**Comment by Godmatik** [comment]
> No I'm not. The odds of my not getting the drop if it's on a 2% drop rate are now so low that you are the one committing the appeal to probability fallacy. We are now at over 2800 kills of this mob with no drop. That's not 2%, it's not anywhere near 2%.

"No I'm not, you are!" 
Lol

**Comment by divinenips** [comment]
> > I'll tell you what bro, if you can get this thing to drop for you and provide screenies, I'll pay you $25 on Venmo. Not even joking. Go for it, prove me wrong.
> 
> I don't need it to drop right now to prove you wrong. There are multiple enchanters in my guild that have had it drop starting from the 2nd week of server up until weeks ago. I even provided evidence from discord with date stamps. Funny thing is, they aren't the only enchanters on the server who have it. There are, most likely, hundreds of enchanters on the server (Horde and Alliance) who have this formula from farming it. Some got it after few kills, others slaved for weeks.
> 
> Seems rather consistent with...
> 
> * many Thottbott/WoWHead comments from 2007/2008
> * my recollection personally farming it multiple times 2007/2008 on retail TBC
> * my recollection personally farming it on THREE other TBC private servers in the last 5 years
> * my recollection personally farming it in 2018/2019 on TBC Netherwing Atlantiss
> 
> So now...You say you've killed 2800 ogres and that seems exaggerated to me, so why don't you provide screenies and data?
> Go for it, prove me wrong.

Why would I be taking screens of my farm? That's just bizarre. As for the comments on Thottbot and everything, there are comments mentioning that this has a higher drop rate than 2%, as well, so are we just ignoring those?

What I'm saying isn't that it hasn't existed on the server, but some interaction recently broke the drop. This was again confirmed by what the GM Crysis found when he looked into the item drops (it hasn't dropped for anybody in several days). You can laugh it up all you want, but there is something wrong with the drop rate on this item and I'm willing to put money where my mouth is. For real, I will give you $25 if you can find the pattern with a recent timestamp on the picture.

**Comment by divinenips** [comment]
> > No I'm not. The odds of my not getting the drop if it's on a 2% drop rate are now so low that you are the one committing the appeal to probability fallacy. We are now at over 2800 kills of this mob with no drop. That's not 2%, it's not anywhere near 2%.
> 
> "No I'm not, you are!"
> Lol

You don't understand probability, at all. If I were to farm for this pattern until the server shut down a year or so from now and it never dropped in that entire time, would you attribute that to "RNG"? What threshold of farming would I have to cross before you would concede that there's an issue here?

**Comment by ScottieDOESknow** [comment]
> Why would I be taking screens of my farm?

If you had proof of farming 2800 ogres with loot data you could provide it here and possibly move this ticket forward. 
If you don't... then it's just guessing and as I suggested, most likely an exaggerated sample size.

> but some interaction recently broke the drop.

like what?

> This was again confirmed by what the GM Crysis found when he looked into the item drops (it hasn't dropped for anybody in several days).

So "something" broke the drop and Crysis confirmed this by telling you that it's hasn't dropped for anyone in a few days? Seems like that only confirms nobody has had it drop in a few days, which is consistent with everything I suggested my last post (low drop chance heavy RNG)

>For real, I will give you $25 if you can find the pattern with a recent timestamp on the picture.

What's the date threshold for recent picture before you would concede that current drop rate is 100% correct? ;)

**Comment by divinenips** [comment]
> > Why would I be taking screens of my farm?
> 
> If you had proof of farming 2800 ogres with loot data you could provide it here and possibly move this ticket forward.
> If you don't... then it's just guessing and as I suggested, most likely an exaggerated sample size.
> 
> > but some interaction recently broke the drop.
> 
> like what?
> 
> > This was again confirmed by what the GM Crysis found when he looked into the item drops (it hasn't dropped for anybody in several days).
> 
> So "something" broke the drop and Crysis confirmed this by telling you that it's hasn't dropped for anyone in a few days? Seems like that only confirms nobody has had it drop in a few days, which is consistent with everything I suggested my last post (low drop chance heavy RNG)
> 
> > For real, I will give you $25 if you can find the pattern with a recent timestamp on the picture.
> 
> What's the date threshold for recent picture before you would concede that current drop rate is 100% correct? ;)

Not sure what the interaction is. The time stamp should be past December 24th. The numbers aren't exaggerated, if anything they are an underestimation. And why exactly do you care, by the way? Why does it matter to you if the devs take a look at the code? What exactly is the point of your arguing this?

**Comment by ScottieDOESknow** [comment]
> The numbers aren't exaggerated, if anything they are an underestimation.

how are you tracking kills of mobs that share spawns with other NPCs? (non-static respawn)
do you have any objective data on your kills/loot?

> Why does it matter to you if the devs take a look at the code?

It doesn't matter to me if they look. In fact, I hope they can confirm for you.

> What exactly is the point of your arguing this?

The clear lack of evidence / data to support your claim initiated my first response. I found your claim inconsistent based on my personal experiences & other friends/guild members experiences farming the same formula between 2007 and 2020. 
It's also inconsistent with the lastest/highest rated WowHead comments from 2007/2008. 

Your claimed sample size started at over 2000 kills in 31 hours and 12 hours later your at over 2800 kills. 
That's a 40% increase (800 kills) in 12 hours time. I suggested an exaggeration of your sample size and I may not be wrong.

**Comment by Wolffenstein** [comment]
1. Not every item drop is logged to the database. GM Crysis provided you with the information that doesn't prove anything since loot generations of this formula are NOT saved in our log system.
2. Currently 426 characters have this formula learnt, so the item definitely drops. If you really killed 2800 Bloodmaul Geomancer without seeing one, you're either unqualified to drop it (360 Enchanting is a requirement here if I'm not mistaken) or you've been massively unlucky.
3. You assume that the wowhead drop chance is lowered due to non-enchanters killing the mobs as well (which is often the case in terms of quest items). If you read the comments, retail players assumed the same and yet still some of them had to kill hundreds of mobs to get it. The key thing to note here is:
a) only people with wowhead addon provide their data to the website 
b) there's not much reason to kill Geomancers outside of farming the recipe (they count towards "The Bloodmaul Ogres" quest but there's plenty of other mobs that are more efficient to kill to complete it)
c) Geomancers were often actively being farmed/hunted by enchanters, which further reduced the chance a non-enchanter would kill them and affect the wowhead ratio while not being able to drop the item)
4. The drop chance on Karazhan is indeed lower than expected, as it's set to 1%. Given the wowhead comments and everything I wrote in point 3, it's very likely the actual drop chance does not deviate from the wowhead chance too much. Still, to account for the possibility of non-enchanters I will set the chance to 3%. Fix will be applied in revision 3580.

**Comment by divinenips** [comment]
> 1. Not every item drop is logged to the database. GM Crysis provided you with the information that doesn't prove anything since loot generations of this formula are NOT saved in our log system.
> 2. Currently 426 characters have this formula learnt, so the item definitely drops. If you really killed 2800 Bloodmaul Geomancer without seeing one, you're either unqualified to drop it (360 Enchanting is a requirement here if I'm not mistaken) or you've been massively unlucky.
> 3. You assume that the wowhead drop chance is lowered due to non-enchanters killing the mobs as well (which is often the case in terms of quest items). If you read the comments, retail players assumed the same and yet still some of them had to kill hundreds of mobs to get it. The key thing to note here is:
>    a) only people with wowhead addon provide their data to the website
>    b) there's not much reason to kill Geomancers outside of farming the recipe (they count towards "The Bloodmaul Ogres" quest but there's plenty of other mobs that are more efficient to kill to complete it)
>    c) Geomancers were often actively being farmed/hunted by enchanters, which further reduced the chance a non-enchanter would kill them and affect the wowhead ratio while not being able to drop the item)
> 4. The drop chance on Karazhan is indeed lower than expected, as it's set to 1%. Given the wowhead comments and everything I wrote in point 3, it's very likely the actual drop chance does not deviate from the wowhead chance too much. Still, to account for the possibility of non-enchanters I will set the chance to 3%. Fix will be applied in revision 3580.

That's all good. 

There is something here that raised a red flag for me, however. When you say "(360 Enchanting is a requirement here if I'm not mistaken)", do you mean to say that the pattern is prohibited from dropping for enchanters below level 360? Because this should not be the case and would constitute a major deviation from retail TBC. Any enchanter of any level should see this pattern drop. Is this not the case?
